TourMate/TourMate/Frontend/Plan/Views/EditPlanViews/EditPlanViewModel.swift

```diff
@@ -6,20 +6,37 @@
 //
 
 import Foundation
+import Combine
 
-class EditPlanViewModel: ObservableObject {
-    @Published private(set) var isLoading: Bool
-    @Published private(set) var hasError: Bool
+class EditPlanViewModel<T: Plan>: ObservableObject {
+    @Published private(set) var isLoading = false
+    @Published private(set) var hasError = false
 
+    @Published var plan: T
+    @Published var isPlanDurationValid = true
+    @Published var canEditPlan = true
+
+    var trip: Trip
```

Comment:
Seems duplicated with `AddPlanFormViewModel`, should we just have one single view model for plans?

Reply:
I think Add can be separated from Fetch,Edit and Delete (something like TripViewModel). Mainly because Adding is done in a separately view, but fetch, edit, delete are usually in a main Trip/PlanView, then we load a sheet to do the update/delete. 
- Merging all into one ViewModel shouldn't be an issue